test/fetch/headers.js

```diff
@@ -719,3 +719,30 @@ test('When the value is updated, update the cache', (t) => {
   headers.append('d', 'd')
   deepStrictEqual([...headers], [...expected, ['d', 'd']])
 })
+
+test('Symbol.iterator is only accessed once', (t) => {
+  const { ok } = tspl(t, { plan: 1 })
+
+  const dict = new Proxy({}, {
+    get () {
+      ok(true)
+
+      return function * () {}
+    }
+  })
+
+  new Headers(dict) // eslint-disable-line no-new
+})
+
+test('Invalid Symbol.iterators', (t) => {
+  const { throws } = tspl(t, { plan: 3 })
+
+  throws(() => new Headers({ [Symbol.iterator]: null }), TypeError)
+  throws(() => new Headers({ [Symbol.iterator]: undefined }), TypeError)
+  throws(() => {
+    const obj = { [Symbol.iterator]: null }
+    Object.defineProperty(obj, Symbol.iterator, { enumerable: false })
+
+    new Headers(obj) // eslint-disable-line no-new
```

Comment:
This throws because Symbol.iterator can't be converted to a string rather than being an invalid dictionary. Technically still a bug, but this also happens in Chrome so I assume no one in the history of mankind will ever notice.

Reply:
As in, we can't implement the spec steps stating to ignore non-enumerable properties because then this case wouldn't throw an error.